### PR TITLE
feat: implement zustand stores for sessions and streams

### LINKED ISSUE #12: Implement Zustand stores for sessions and messages
State management for session list, active session, messages, and streaming state.

Ref: ADR 004
Depends on: #5

### PR DESCRIPTION
**Pull request by vriesdemichael**
Implements session, messages, and streaming event states using Zustand and Immer for Issue #12. Fixes #12.

### CODE REVIEW

File: app/store/__tests__/session-test.ts [hunk 1]
```diff
@@ -364,6 +364,87 @@ describe("Session Store", () => {
...
+
+					act(() =>
+						useSessionStore
+							.getState()
+							.onSessionStatus(sessionId, { status: "active" }),
+					);
+
+					const state = useSessionStore.getState();
+					expect(state.sessions[0].status).toEqual({ status: "active" });
+				});
```
[the remark this answers is not in the document]
Comment (vriesdemichael) [reply app/store/__tests__/session-test.ts]: Fixed: Added edge case test for missing session in onSessionStatus.
